### DISCUSSION TITLE
Ender 3 V2

### DISCUSSION THREAD

**Discussion by molesza** [Sakati84/3DPrintColorizer#6] [General]
Such a clever idea!

Super excited to try this out. I have an Ender 3 v2. Going to print and make adjustments to the holder as needed. Once I'm done I'll post a link to the ender 3 v2 stl.

**Comment by Sakati84** [comment]
That would be absolutly awsome!

Best regards,
Andreas

molesza ***@***.***> schrieb am Do., 22. Juli 2021, 14:05:

**Reply by hujiba** [reply]
Great! I was really looking forward to using this on my V2

**Reply by molesza** [reply]
Available now! https://www.thingiverse.com/thing:4921206

**Comment by molesza** [comment]
Its done!

https://www.thingiverse.com/thing:4921206

Its not the best design but it works! Maybe someone else can clean it up a bit.

**Reply by molesza** [reply]
used the bltouch mount and modified.
https://www.thingiverse.com/thing:4462870

**Reply by Sakati84** [reply]
Very cool! Would you mind if i put it in the GitHub Repo. You would be given credit of course.

**Reply by molesza** [reply]
> Very cool! Would you mind if i put it in the GitHub Repo. You would be given credit of course.

Please do!

**Reply by csanii123** [reply]
I made an adapter for the Ender 3 v2 as well for those without a BLtouch, but probably I over-engineered it. I am adjusting the height of the pen holder after you home the nozzle. I thought I had it "right" but I must be too low as one of the markers is able to touch the bed when it is on the 1st layer or 2 of the print. What is your measurement from bottom of the pen holder to the bed right after you level the bed ? Mine is 5.5mm but that must be too close. as 1 pen is able to write on the bed while the other colors fine on the print.

**Reply by Sakati84** [reply]
The tip of the holder is 3.2 mm above the ground for me. The pen should be able to write on the bed - that's fine! When you calibrate everything correctly it will only draw directly on the printed layer. If you can´t calibrate it correctly you should print with a skirt so it doesn´t color the print bed. If you get paint on the print bed remove it with alcohol.

**Reply by csanii123** [reply]
I triple checked my settings. The tip of the holder is 3.9mm above the bed for me. When the pen moves across the bed it writes on the bed but then it hops up and over to get to the item to color it in. It then colors the item then lowers down again to go across the bed thus painting the bed again and then goes up to swap pens. Please note this is before I reduce the Z-offset by the recommended 2-2.5mm. If I reduce the Z-offset then I am sure it will really be coloring the bed. If the pen didn't hop over the print I would say no issue but the pen actually going up and over the print makes me think there may be a bug in the code for the 1st layers. On higher layers it doesn't matter as the pen won't be close to the bed anyways. Could you take a look at the code and see if after it lays down the 1st layer that it increases the height so that it doesn't write on the bed ? For the 1st few layers it likely should print the layer then go up and grab the pen and then on its way down shift the X and Y and then the Z so that it lands directly on the print vs going across the bed. Similarly when it is done coloring it should move up the Z a bit then do the X and Y so that again it doesn't paint the bed. Hopefully this makes sense...
